controllers/shipwrightbuild_controller.go

```diff
@@ -69,7 +79,75 @@ func (r *ShipwrightBuildReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 	logger := r.Logger.WithValues("namespace", req.Namespace, "name", req.Name)
 	logger.Info("Starting resource reconciliation...")
 
-	// retrieving the ShipwrightBuild instance requested for reconciliation
+	// See if tekton is there
+	_, tektonAPIErr := r.CRDClient.CustomResourceDefinitions().Get(ctx, "taskruns.tekton.dev", metav1.GetOptions{})
+	tektonOpCRD, tektonOPErr := r.CRDClient.CustomResourceDefinitions().Get(ctx, "tektonconfigs.operator.tekton.dev", metav1.GetOptions{})
+	switch {
+	case tektonOPErr == nil && tektonAPIErr == nil:
+		logger.Info("Tekton is installed, and the Tekton Operator is also installed.")
+	case tektonAPIErr == nil && tektonOPErr != nil:
+		logger.Info("Tekton has been installed without use of its associated operator.")
+	case errors.IsNotFound(tektonAPIErr) && errors.IsNotFound(tektonOPErr):
+		//TODO should we error out here or proceed and give the user the opportunity of install Tekton after Shipwright?
+		logger.Info("Evidence of neither required Tekton APIs nor the Tekton Operator being installed can be found.  Will proceed with the Shipwright installation attempt, but Shiipwright will not be functional without Tekton")
+	case tektonAPIErr != nil && tektonOPErr == nil:
+		if tektonOpCRD.Labels == nil {
+			retErr := fmt.Errorf("the CRD TektonConfig does not have labels set, inclding its version")
+			logger.Error(retErr, "Problem confirming Tekton Operator version")
+			return RequeueWithError(retErr)
+		}
+		value, exists := tektonOpCRD.Labels["version"]
+		if !exists {
+			retErr := fmt.Errorf("the CRD TektonConfig does not have labels set, inclding its version")
+			logger.Error(retErr, "Problem confirming Tekton Operator version")
+			return RequeueWithError(retErr)
```

Comment:
Similar - this is a permanent failure.

Reply:
will change to perm fail / error out in next push